docs/dev/version-migration/migration-guide.md

```diff
@@ -0,0 +1,22 @@
+# `partiql-lang-kotlin` Version Migration Guide
+
+(Note: this folder's documents will likely be converted to a set of wikis on https://github.com/partiql/partiql-lang-kotlin/wiki
+with each major version bump on a separate wiki page.
+TBD if these documents will remain as a markdown file on GitHub)
+
+The `partiql-lang-kotlin` [release notes](https://github.com/partiql/partiql-lang-kotlin/releases) list changes 
+between minor versions as well as the last minor version to the subsequent major version (e.g. v0.2.7 -> v0.3.0). This 
+document lists the aggregated release notes between major `partiql-lang-kotlin` versions (e.g. v0.2.7 -> v0.3.4). To 
+make it easier to understand differences between `partiql-lang-kotlin` major versions, we've also
+* unified the change format (new features, breaking changes - behavioral and API, deprecated items, misc/bug fixes)
+* cleaned up some release note items (summarized sequences of related commits, omitted commits related to tests and 
+build-related changes)
+* moved CLI/REPL changes to a separate section
+* found other breaking changes from the compatibility reports between versions using the [japi-compliance-checker](https://github.com/lvc/japi-compliance-checker)
+
+The repo, [test-partiql-version-migration](https://github.com/alancai98/test-partiql-version-migration), hosts all the 
```

Comment:
Slightly unrelated: shouldn't be `test-partiql-version-migration` be `partiql-test-version-migration`?

Reply:
I see, it's not, but since all of our public repos start w/ `partiql-` I thought haven't named consistently makes sense.

Reply:
Since it's a private repo, I don't have a strong opinion on this.

Reply:
Curious, are we planning to make your personal repo, public?

Reply:
The repo is currently public. Perhaps for future maintenance and updating, it would be better to move it under the `partiql` organization (either as a separate repo or a submodule for `partiql-lang-kotlin`).

Reply:
Oh "bad wording", by "public" I meant under the `partiql` org. Can we have a `TODO` for that?

Reply:
Agree on moving it to the `partiql` org. Created an issue to track: https://github.com/partiql/partiql-lang-kotlin/issues/692 and added a TODO.